apps/openmw/mwsound/soundmanagerimp.cpp

```diff
@@ -554,13 +554,13 @@ namespace MWSound
         mActiveSaySounds.insert(std::make_pair(MWWorld::ConstPtr(), sound));
     }
 
-    bool SoundManager::sayDone(const MWWorld::ConstPtr &ptr) const
+    bool SoundManager::sayDone(const MWWorld::ConstPtr &ptr, bool ignoreInit) const
     {
         SaySoundMap::const_iterator snditer = mActiveSaySounds.find(ptr);
         if(snditer != mActiveSaySounds.end())
         {
             if(mOutput->isStreamPlaying(snditer->second))
-                return false;
+                return ignoreInit && !mOutput->getStreamOffset(snditer->second);
```

Comment:
This doesn't look safe since the stream offset will increment whenever OpenAL processes some audio, which can be any time after the stream playback starts. That would cause `SayDone` to return 0 randomly on the same frame `Say` was called, depending on when the next internal mixing update is scheduled.

Reply:
Ok, I had another idea. Here's something somewhat different (no doubt equally crutchy).